librz/core/canalysis.c

```diff
@@ -6726,4 +6726,130 @@ RZ_API bool rz_core_analysis_esil_trace_stop(RzCore *core) {
 	esil->trace = NULL;
 	rz_config_set_i(core->config, "dbg.trace", false);
 	return true;
-}
\ No newline at end of file
+}
+
+/**
+ * Free RzAnalysisBytes
+ *
+ * @param ptr RzAnalysisBytes pointer
+ */
+RZ_API void rz_analysis_bytes_free(void *ptr) {
+	rz_return_if_fail(ptr);
+	RzAnalysisBytes *ab = ptr;
+	rz_analysis_op_free(ab->op);
+	ab->op = NULL;
+	rz_analysis_hint_free(ab->hint);
+	ab->hint = NULL;
+	RZ_FREE(ab->opcode);
+	RZ_FREE(ab->disasm);
+	RZ_FREE(ab->pseudo);
+	RZ_FREE(ab->description);
+	RZ_FREE(ab->mask);
+	RZ_FREE(ab->bytes);
+	RZ_FREE(ab);
+}
+
+/**
+ *
+ * Analysis and disassemble bytes use rz_analysis_op and rz_asm_disassemble
+ *
+ * @param core The RzCore instance
+ * @param buf data to analysis
+ * @param len analysis len bytes
+ * @param nops analysis n ops
+ * @return list of RzAnalysisBytes
+ */
+RZ_API RZ_OWN RzList *rz_core_analysis_bytes(RzCore *core, const ut8 *buf, int len, int nops) {
+	bool be = core->print->big_endian;
+	core->parser->subrel = rz_config_get_i(core->config, "asm.sub.rel");
+	int ret, i, idx;
+	RzAsmOp asmop;
+	ut64 addr;
+	RzList *list = rz_list_newf(rz_analysis_bytes_free);
```

Comment:
check for null, also since you know the numbe of ops to disassemble, then why not just use RzPVector?